src/pudl/analysis/clean_up_ferc1.py

```diff
@@ -0,0 +1,655 @@
+"""Clean up the data from FERC Form 1.
+
+The data from FERC Form 1 is notoriously bad. The small generators table, for example,
+contains rows that represent different types of information: headers, values, and notes.
+It's like someone took a paper form, copied it into Microsoft Excel, and called it a
+day. Needless to say, it needs a lot of help before it can be used programatically for
+bulk analysis.
```

Comment:
Is it possible to provide some examples of raw small generators records and the desired outputs after the transformations are applied?

Reply:
done!